src/Forms/Shared/Samples/Data/EditBranchVersioning/EditBranchVersioning.xaml.cs

```diff
@@ -0,0 +1,361 @@
+
+                    // Update the combobox selection without triggering the event.
+                    DamageBox.SelectedItem = currentAttributeValue;
+
+                    MoveText.IsVisible = DamageBox.IsEnabled = _serviceGeodatabase.VersionName != _serviceGeodatabase.DefaultVersionName;
+                }
+                catch (Exception ex)
+                {
+                    ShowAlert(ex.Message, ex.GetType().Name);
+                }
+                finally
+                {
+                    // Update the UI for the selection.
+                    SwitchView(AttributeView);
+                }
+            }
+        }
+
+        private async Task ApplyDamageChange()
+        {
+            try
+            {
+                // Get the value from the UI.
+                string selectedAttributeValue = DamageBox.SelectedItem.ToString();
+
+                // Check if the new value is the same as the existing value.
+                if (_selectedFeature.Attributes[_attributeFieldName].ToString() == selectedAttributeValue)
+                {
+                    return;
+                }
+
+                // Load the feature.
+                await _selectedFeature.LoadAsync();
+
+                // Update the attribute value.
+                _selectedFeature.SetAttributeValue(_attributeFieldName, selectedAttributeValue);
+
+                // Update the table.
+                await _selectedFeature.FeatureTable.UpdateFeatureAsync(_selectedFeature);
+
+                // Update the service.
+                await ((ServiceFeatureTable)_selectedFeature.FeatureTable).ApplyEditsAsync();
+
+                SwitchView(DefaultView);
+                ShowAlert("Edited feature " + _selectedFeature.Attributes["objectid"]);
+
+                // Clear the selection.
+                _featureLayer.ClearSelection();
+                _selectedFeature = null;
+            }
+            catch (Exception ex)
+            {
+                ShowAlert(ex.Message, "Failed to edit feature");
+            }
+        }
+
+        private async void ConfirmVersionClick(object sender, EventArgs e)
+        {
+            try
+            {
+                // Validate name and access input.
+                if (!VersionNameValid(NameEntryBox.Text))
+                {
+                    return;
+                }
+
+                if (!(AccessBox.SelectedItem is VersionAccess))
+                {
+                    ShowAlert("Please select an access level");
+                    return;
+                }
+
+                // Set the user defined name, access level and description as service version parameters
+                ServiceVersionParameters newVersionParameters = new ServiceVersionParameters();
+                newVersionParameters.Name = NameEntryBox.Text;
+                newVersionParameters.Access = (VersionAccess)AccessBox.SelectedItem;
+                newVersionParameters.Description = DescriptionBox.Text;
```

Comment:
Interestingly, on Forms UWP, if the user leaves the description field blank, an ArgumentNullException is thrown here. Apparently the textbox text defaults to null rather than an empty string. Curiously, the argument null exception is thrown when the description is set, but not if the description is never set. I recommend checking for null here and using an empty string if it is `= DescriptionBox.Text ?? "";` or similar.